src/OrleansTestKit/Storage/StorageExtensions.cs

```diff
@@ -1,29 +1,111 @@
 ﻿using System;
 using System.ComponentModel;
+using System.Reflection;
+using System.Threading.Tasks;
+using Moq;
+using Orleans.Core;
+using Orleans.Runtime;
+using Orleans.Storage;
 using Orleans.TestKit.Storage;
 
 namespace Orleans.TestKit
 {
     public static class StorageExtensions
     {
-        public static TState State<TState>(this TestKitSilo silo) where TState : class, new()
+        public static TState State<TGrain, TState>(this TestKitSilo silo)
+            where TGrain : Grain<TState>
         {
             if (silo == null)
             {
                 throw new ArgumentNullException(nameof(silo));
             }
 
-            return silo.StorageManager.GetStorage<TState>().State;
+            return silo.StorageManager.GetGrainStorage<TGrain, TState>().State;
         }
 
-        public static TestStorageStats StorageStats(this TestKitSilo silo)
+        public static TestStorageStats StorageStats<TGrain, TState>(this TestKitSilo silo)
+            where TGrain : Grain<TState>
         {
             if (silo == null)
             {
                 throw new ArgumentNullException(nameof(silo));
             }
 
-            return silo.StorageManager.StorageStats;
+            return silo.StorageManager.GetStorageStats<TGrain, TState>();
+        }
+
+        public static IStorage<T> AddGrainState<TGrain, T>(
+            this TestKitSilo silo,
+            T state = default)
+            where TGrain : Grain<T>
```

Comment:
Isn't `Grain<T>` effectively deprecated? Wouldn't `TGrain : Grain` be a more flexible fit here? Perhaps I'm missing the context...

Reply:
Yup these methods are explictly for old school grain state.